pdns/ueberbackend.cc

```diff
@@ -632,36 +678,65 @@ void UeberBackend::getAllDomains(vector<DomainInfo> *domains, bool include_disab
 
 bool UeberBackend::get(DNSZoneRecord &rr)
 {
-  // cout<<"UeberBackend::get(DNSZoneRecord) called"<<endl;
-  if(d_negcached) {
+  // cout<<"UeberBackend::get(DNSZoneRecord) called for "<<d_question.qname.toString()<<" / "<<d_question.qtype.getName()<<endl;
+  if (d_negcached) {
     return false; 
   }
 
-  if(d_cached) {
-    if(d_cachehandleiter != d_answers.end()) {
-      rr=*d_cachehandleiter++;;
+  if (d_cached) {
+    if (d_cachehandleiter != d_answers.end()) {
+      rr = *d_cachehandleiter++;
```

Comment:
I think we need a check here to make sure the zone_id in rr matches the id passed to lookup()

Reply:
QC.getEntry already uses the zoneId for the cache lookup, no?